modules/tensor_mechanics/src/materials/CappedMohrCoulombCosseratStressUpdate.C

```diff
@@ -0,0 +1,120 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "CappedMohrCoulombCosseratStressUpdate.h"
+#include "libmesh/utility.h"
+
+template <>
+InputParameters
+validParams<CappedMohrCoulombCosseratStressUpdate>()
+{
+  InputParameters params = validParams<CappedMohrCoulombStressUpdate>();
+  params.addClassDescription("Capped Mohr-Coulomb plasticity stress calculator for the Cosserat "
+                             "situation where the host medium (ie, the limit where all Cosserat "
+                             "effects are zero) is isotropic.  Note that the return-map flow rule "
+                             "uses an isotropic elasticity tensor built with the 'host' properties "
+                             "defined by the user.");
+  params.addRequiredRangeCheckedParam<Real>("host_youngs_modulus",
+                                            "host_youngs_modulus>0",
+                                            "Young's modulus for the isotropic host medium");
+  params.addRequiredRangeCheckedParam<Real>("host_poissons_ratio",
+                                            "host_poissons_ratio>=0",
+                                            "Poisson's ratio for the isotropic host medium");
+  return params;
+}
+
+CappedMohrCoulombCosseratStressUpdate::CappedMohrCoulombCosseratStressUpdate(
+    const InputParameters & parameters)
+  : CappedMohrCoulombStressUpdate(parameters),
+    _host_young(getParam<Real>("host_youngs_modulus")),
+    _host_poisson(getParam<Real>("host_poissons_ratio")),
+    _host_E0011(_host_young * _host_poisson / (1.0 + _host_poisson) / (1.0 - 2.0 * _host_poisson)),
+    _host_E0000(_host_E0011 + _host_young / (1.0 + _host_poisson))
+{
+}
+
+void
+CappedMohrCoulombCosseratStressUpdate::preReturnMapV(
+    const std::vector<Real> & /*trial_stress_params*/,
+    const RankTwoTensor & stress_trial,
+    const std::vector<Real> & /*intnl_old*/,
+    const std::vector<Real> & /*yf*/,
+    const RankFourTensor & /*Eijkl*/)
+{
+  std::vector<Real> eigvals;
+  stress_trial.symmetricEigenvaluesEigenvectors(eigvals, _eigvecs);
+  _poissons_ratio = _host_poisson;
+}
+
+void
+CappedMohrCoulombCosseratStressUpdate::setEffectiveElasticity(const RankFourTensor & /*Eijkl*/)
+{
+  _Eij[0][0] = _Eij[1][1] = _Eij[2][2] = _host_E0000;
+  _Eij[0][1] = _Eij[1][0] = _Eij[0][2] = _Eij[2][0] = _Eij[1][2] = _Eij[2][1] = _host_E0011;
+  _En = _Eij[2][2];
+  const Real denom = _Eij[0][0] * (_Eij[0][0] + _Eij[0][1]) - 2 * Utility::pow<2>(_Eij[0][1]);
```

Comment:
let's put an assertion here for `demon > LIBMESH_TOL`

Reply:
wait, where? I don't see the assertion.